src/v/redpanda/debug_bundle.cc

```diff
@@ -81,7 +82,11 @@ debug_bundle::debug_bundle(
   : _status{debug_bundle_status::not_running}
   , _write_dir{write_dir}
   , _in_progress_filename{"debug-bundle.zip"}
-  , _rpk_cmd{rpk_path} {}
+  , _rpk_cmd{rpk_path}
+  , _transfer_handle{
+      detail::make_bundle_filename(get_write_dir(), _in_progress_filename),
+      nullptr,
+      false} {}
```

Comment:
So, I wanted to created this within `fetch_bundle`; however, runtime problems occurred because `transfer_handle` was destroyed before the multi-part upload could complete. This is an important aspect about `ss::experimental::process` and `file_handler`. The handler **must** remain live during the whole multi-part upload.

So I put the handler as a class object but when we support multiple bundles on disk, I will likely have to do one of two things:
1. Turn the file handler into a pointer and construct it on the fly
2. Refactor `ss::httpd::file_handler` to have a way to se the filename

I'm leaning towards (2)

Reply:
Maybe I'm being dense, but can't you declare a `ss::http::file_handler` within `fetch_bundle` and the use a `ss::do_with` to maintain it?

Reply:
Oh I forgot about `do_with`. I'll try that out.